packages/util-waiter/src/poller.ts

```diff
@@ -0,0 +1,36 @@
+import { sleep } from "./utils/sleep";
+import { WaiterOptions, WaiterResult, WaiterState } from "./waiter";
+
+/**
+ * Reference: https://github.com/awslabs/smithy/pull/656
```

Comment:
I think it's better to link to the waiters spec rather than the PR to update the spec. Maybe link here since this link will be relevant when we release the next version of Smithy that includes jitter in waiters: https://awslabs.github.io/smithy/1.0/spec/waiters.html#waiter-retries